controllers/common/step.go

```diff
@@ -24,5 +24,5 @@ import (
 )
 
 func AllSteps() []pipeline.PipelineStep {
-	return []pipeline.PipelineStep{finalizers.Step, desiredphase.Step, condition.Step, records.Step}
+	return []pipeline.PipelineStep{finalizers.Step, desiredphase.Step, condition.Step, records.Step, condition.Step}
 }
```

Comment:
@STRRL  when I develop  this patch, i tried to remove the first `condition `, but it will be a bug. when we pause a experiment(which keep failing to inject), the object will be requeued in the `records` step, and not enter the `condition` step. then, the `Paused` condition will still be false...

so, I just add another `condition` step after the `records` step. and did not remove the first one.

Reply:
Maybe this change is not required after we do not use the Status Subresrouce